### PR TITLE
HEEDLS-323 Content viewer smaller height on mobile

### PR DESCRIPTION
**Pull request by kwunkiuma**
Add wrapper for responsive sizing on iframe. The iframe now maintains a 4:3 ratio.

Change fullscreen scripts.

Modify fullscreen tests.

Issue with padding on sides is also resolved.

Ran and passed all unit tests, screenshots taken in Google Chrome.

Browser size
image

Mobile size
image

Issue on Firefox where the height of content inside the iframe is too big.
image

### CODE REVIEW

File: DigitalLearningSolutions.Web/Views/LearningMenu/Tutorial/ContentViewer.cshtml [hunk 1]
```diff
@@ -39,20 +39,22 @@
...
-    <a class="nhsuk-button js-only-inline"
-       id="content-viewer_enter-fullscreen-button"
-       href="#">
-      Fullscreen
-    </a>
   </div>
 </div>
+  
+@{ var contentType = "tutorial"; }
+<partial name="Shared/_JavaScriptDisabledError" model="contentType" />
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web/Views/LearningMenu/Tutorial/ContentViewer.cshtml:46]: Does it still look okay without js? 🙂

File: DigitalLearningSolutions.Web/Views/LearningMenu/Tutorial/ContentViewer.cshtml [hunk 2]
```diff
@@ -39,20 +39,22 @@
...
+
+<div class="nhsuk-u-margin-bottom-6 content-viewer_wrapper" id="content-viewer_wrapper">
+  <iframe src="@Model.ContentSource" class="nhsuk-u-margin-bottom-6 js-only-block content-viewer_iframe" id="content-viewer_iframe">
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web/Views/LearningMenu/Tutorial/ContentViewer.cshtml:49]: Does this still need an id? 🙂

File: DigitalLearningSolutions.Web/Scripts/spec/learningMenu/fullscreen.spec.ts [hunk 3]
```diff
@@ -105,18 +105,16 @@ describe('enterFullscreen', () => {
...
       <head></head>
       <body>
-        <iframe
-          src="https://www.dls.nhs.uk/CMS/CMSContent/Course324/Section1058/Tutorials/01-01-CreateaPresentation/itspplayer.html?CentreID=101&amp;CustomisationID=19068&amp;TutorialID=4671&amp;CandidateID=254480&amp;Version=2&amp;ProgressID=285049&amp;type=learn&amp;TrackURL=https://localhost:44367/tracking/tracker"
-          class="nhsuk-grid-column-full nhsuk-u-margin-bottom-6 js-only-block"
-          height="800"
-          id="content-viewer_iframe">
-        </iframe>
+        <div class="nhsuk-u-margin-bottom-6 content-viewer_wrapper" id="content-viewer_wrapper">
+          <iframe src="https://www.dls.nhs.uk/CMS/CMSContent/Course324/Section1058/Tutorials/01-01-CreateaPresentation/itspplayer.html?CentreID=101&amp;CustomisationID=19068&amp;TutorialID=4671&amp;CandidateID=254480&amp;Version=2&amp;ProgressID=285049&amp;type=learn&amp;TrackURL=https://localhost:44367/tracking/tracker" class="nhsuk-u-margin-bottom-6 js-only-block content-viewer_iframe" id="content-viewer_iframe">
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web/Scripts/spec/learningMenu/fullscreen.spec.ts:115]: Minor but I'd suggest splitting this over several lines like it was before, to make it easier to read 🙂

File: DigitalLearningSolutions.Web/Scripts/spec/learningMenu/fullscreen.spec.ts [hunk 4]
```diff
@@ -125,8 +123,8 @@ describe('enterFullscreen', () => {
     enterFullscreen();
 
     // Then
-    const iframe = document.getElementById('content-viewer_iframe');
-    expect(iframe!.classList).toContain('fullscreen');
+    const wrapper = document.getElementById('content-viewer_wrapper');
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web/Scripts/spec/learningMenu/fullscreen.spec.ts:126]: Minor but I think `content-viewer_iframe-wrapper` is a more descriptive name 🙂

**Review by FridaTveit — COMMENTED** [review]
A few comments, otherwise looks good 🙂

File: DigitalLearningSolutions.Web/Views/LearningMenu/Tutorial/ContentViewer.cshtml (hunk 1 above)
Comment (kwunkiuma) [reply DigitalLearningSolutions.Web/Views/LearningMenu/Tutorial/ContentViewer.cshtml:46]: The error message is being shown as expected.

File: DigitalLearningSolutions.Web/Scripts/spec/learningMenu/fullscreen.spec.ts (hunk 4 above)
Comment (kwunkiuma) [reply DigitalLearningSolutions.Web/Scripts/spec/learningMenu/fullscreen.spec.ts:126]: Would you like me to rename the class as well?

Comment (FridaTveit) [reply DigitalLearningSolutions.Web/Scripts/spec/learningMenu/fullscreen.spec.ts:126]: Yes please 🙂

File: DigitalLearningSolutions.Web/Scripts/learningMenu/fullscreen.ts [hunk 5]
```diff
@@ -10,7 +10,8 @@ export function setupFullscreen(): void {
 }
 
 export function exitFullscreen(): void {
-  getIFrame()?.classList.remove('fullscreen');
+  getIFrameWrapper()?.classList.remove('fullscreen');
+  getIFrameWrapper()?.classList.add('content-viewer_iframe-wrapper');
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web/Scripts/learningMenu/fullscreen.ts:14]: We should test this class being added and removed 🙂

**Review by FridaTveit — COMMENTED** [review]
One small thing I didn't consider before, otherwise looks good 🙂

File: DigitalLearningSolutions.Web/Scripts/learningMenu/fullscreen.ts (hunk 5 above)
Comment (kwunkiuma) [reply DigitalLearningSolutions.Web/Scripts/learningMenu/fullscreen.ts:14]: Would you like separate tests for these or should I add the check to the existing tests?

Comment (FridaTveit) [reply DigitalLearningSolutions.Web/Scripts/learningMenu/fullscreen.ts:14]: Separate tests please. Each test should check just one thing 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**